nodejs/eks/nodegroup.ts

```diff
@@ -652,19 +653,54 @@ export function createNodeGroup(
     return createNodeGroupInternal(name, args, pulumi.output(core), parent, provider);
 }
 
+export function resolveOrGetInstanceProfile(
+    name: string,
+    args: Omit<NodeGroupOptions, "cluster">,
+    c: pulumi.UnwrappedObject<CoreData>,
+    parent: pulumi.ComponentResource,
+    provider?: pulumi.ProviderResource,
+): aws.iam.InstanceProfile {
+    if (
+        !args.instanceProfile &&
+        !args.instanceProfileName &&
+        !c.nodeGroupOptions.instanceProfile &&
+        !c.nodeGroupOptions.instanceProfileName
+    ) {
+        throw new pulumi.ResourceError(
+            `an instanceProfile or instanceProfileName is required`,
+            parent,
+        );
+    }
+    if (
+        (args.instanceProfile || c.nodeGroupOptions.instanceProfile) &&
+        (args.instanceProfileName || c.nodeGroupOptions.instanceProfileName)
+    ) {
+        throw new pulumi.ResourceError(
+            `invalid args for node group ${name}, instanceProfile and instanceProfileName are mutually exclusive`,
+            parent,
+        );
+    }
+    if (args.instanceProfile || c.nodeGroupOptions.instanceProfile) {
+        return args.instanceProfile ?? c.nodeGroupOptions.instanceProfile!;
+    }
+    return aws.iam.InstanceProfile.get(
```

Comment:
I think pulling in the resource with the `.get` function would incur RUM costs. We circumvent this by using the data source instead.

It seems like we're only interested in the ARN, right? Instance Profile ARNs follow this format: `arn:aws:iam::<account-id>:instance-profile/<profile-name>`.
If we know the name we can construct it without having to load the resource.

In that case we need other props, we could use [`iam.getInstanceProfileOutput`](https://www.pulumi.com/registry/packages/aws/api-docs/iam/getinstanceprofile/) instead.